### PR TITLE
remove custom_dimension_filter feature and views_per_visit_metric flags

### PR DESCRIPTION
**Pull request by ruslandoga**
### Changes

This PR enables custom dimension filter for self-hosters.

Relevant discussion: https://github.com/plausible/analytics/discussions/2984

This PR is based on https://github.com/plausible/analytics/pull/2898

### Tests
- [x] This PR does not require tests

### Changelog
- [x] Entry has been added to changelog

### Documentation
- [x] This change does not need a documentation update

### Dark mode
- [ ] The UI has been tested both in dark and light mode

### CODE REVIEW

File: lib/plausible_web/controllers/stats_controller.ex [hunk 1]
```diff
@@ -326,7 +326,6 @@ defmodule PlausibleWeb.StatsController do
 
   defp get_flags(user) do
     %{
-      custom_dimension_filter: FunWithFlags.enabled?(:custom_dimension_filter, for: user),
       views_per_visit_metric: FunWithFlags.enabled?(:views_per_visit_metric, for: user)
```
Comment (ruslandoga) [inline lib/plausible_web/controllers/stats_controller.ex:329]: Can this be enabled as well?

https://github.com/plausible/analytics/discussions/131#discussioncomment-6052701

**Comment by ukutaht** [comment]
@RobertJoonas could you take a look? I think you're most familiar with this stuff

File: assets/js/dashboard/filters.js [hunk 2]
```diff
@@ -83,10 +79,6 @@ function renderDropdownFilter(site, history, [key, value], query) {
   )
 }
 
-function shouldLinkToFilterModal(site, key) {
-  return key !== 'props' || site.flags.custom_dimension_filter
```
Comment (aerosol) [inline assets/js/dashboard/filters.js:87]: I lack context here but sanity check -- shouldn't we retain the `key !== 'props'` condition?

**Review by aerosol — APPROVED** [review]

File: assets/js/dashboard/filters.js (hunk 2 above)
Comment (ruslandoga) [reply assets/js/dashboard/filters.js:87]: I lack context here too. My thinking was, if `custom_dimension_filter` is enabled for all, then it's always `true`, so even if `key !== 'props'` returns `false`, the end result is still `true`.

**Comment by aerosol** [comment]
@ruslandoga could you resolve the CHANGELOG conflict please?

File: assets/js/dashboard/filters.js (hunk 2 above)
Comment (RobertJoonas) [reply assets/js/dashboard/filters.js:87]: The `key !== 'props'` check makes sure that if `key` is anything other than 'props' (e.g. 'page') then it should return `true`, no matter the flag status. We can remove that too as we don't need any special handling for props anymore, all of the options should be links.

File: lib/plausible_web/controllers/stats_controller.ex (hunk 1 above)
Comment (RobertJoonas) [reply lib/plausible_web/controllers/stats_controller.ex:329]: > Can views_per_visit_metric be enabled as well?

@ruslandoga, the `views_per_visit_metric` flag is actually redundant. It's a duplicate that can be safely deleted now.

We used a `visits_metric` flag for both `visits` and `views_per_visits` metrics. And that flag was removed in https://github.com/plausible/analytics/pull/2898, so in the current master the `views_per_visit` metric is enabled for everyone.

**Review by RobertJoonas — APPROVED** [review]
Looks good to me!

File: assets/js/dashboard/filters.js [hunk 3]
```diff
@@ -55,26 +55,22 @@ function renderDropdownFilter(site, history, [key, value], query) {
...
+        <Link
+          title={`Edit filter: ${formattedFilters[key]}`}
+          to={{ pathname: `/${encodeURIComponent(site.domain)}/filter/${filterGroupForFilter(key)}`, search: window.location.search }}
+          className="group flex w-full justify-between items-center"
+          style={{ width: 'calc(100% - 1.5rem)' }}
+        >
+          <span className="inline-block w-full truncate">{filterText(key, value, query)}</span>
+          <PencilSquareIcon className="w-4 h-4 ml-1 cursor-pointer group-hover:text-indigo-700 dark:group-hover:text-indigo-500" />
+        </Link>
+        <div
```
Comment (ruslandoga) [inline assets/js/dashboard/filters.js:67]: Oops, this shouldn't be included, since it was using a `!shouldLinkToFilterModal(site, key)` check.

Comment (ruslandoga) [reply assets/js/dashboard/filters.js]: Fixed in https://github.com/plausible/analytics/pull/2996/commits/bc7a61640bb77f8520ba6219a98021eaa7120408

File: lib/plausible_web/controllers/stats_controller.ex (hunk 1 above)
Comment (ruslandoga) [reply lib/plausible_web/controllers/stats_controller.ex:329]: PR into this branch: https://github.com/plausible/analytics/pull/3001

Comment (RobertJoonas) [reply lib/plausible_web/controllers/stats_controller.ex]: @ruslandoga nice! I've approved :)

**Review by RobertJoonas — APPROVED** [review]

**Pull request merged**